### PR TITLE
Add github actions workflows

### PR DESCRIPTION
**Pull request by n8225**
Added monthly checks workflow, with a schedule to run on the 1st of the month, can also be manually ran. On completion a new issue will be created with a link to the job.  Right now the output of check-github-commit-dates.py is the only thing being posted on the issue.  Will need to update test.js to output markdown as well as a script to do that for awesome_bot. Results are also uploaded as artifacts.

I think we should also look into alternatives to awesome_bot.  https://github.com/lycheeverse/lychee looks promising, (fast, ability to use regex to prevent links checked in the python script).

Added test README.md for testing to prevent API rate limiting.

Added checks on PR.

Rewrote part of check-github-commit-dates.py, it now calls the GitHub graphql API directly. It doesn't hit the rate limit. This could use some more error checking.

@nodiscc it looks like actions isn't enabled for the repo.

#1852

### CODE REVIEW

File: .github/workflows/monthly.yml [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+        uses: actions/setup-node@v1
+        with:
+          node-version: '14.x'
+      - name: Setup Checks
+        run: |
+          cd tests 
+          npm install --silent chalk
+          cd ..
+      - name: Set env
+        run: echo "FILE=${{ github.event.inputs.file || 'README.md' }}" >> $GITHUB_ENV;
```
Comment (nodiscc) [inline .github/workflows/monthly.yml:51]: should always be `echo "FILE='README.md' >> $GITHUB_ENV; ` but I think this is redundant anyway? Just use `github.event.inputs.file` in the later steps?

File: .github/workflows/monthly.yml [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+    steps:
+      - uses: actions/checkout@v2
+      - name: Set up Ruby 2.6
+        uses: ruby/setup-ruby@v1
+        with:
+          ruby-version: 2.6.7
+      - name: Setup Checks
+        run: gem install awesome_bot
+      - name: Set env
+        run: echo "FILE=${{ github.event.inputs.file || 'README.md' }}" >> $GITHUB_ENV
```
Comment (nodiscc) [inline .github/workflows/monthly.yml:72]: same as above

File: .github/workflows/monthly.yml [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+    steps:
+      - uses: actions/checkout@v2
+      - name: Setup Python 3.x
+        uses: actions/setup-python@v2
+        with:
+          python-version: '3.x'
+      - name: Setup Checks
+        run: pip3 install Requests
+      - name: Set env
+        run: echo "FILE=${{ github.event.inputs.file || 'README.md' }}" >> $GITHUB_ENV
```
Comment (nodiscc) [inline .github/workflows/monthly.yml:98]: same as above

File: .github/workflows/pull_request.yml [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Set up Ruby 2.6
+        uses: actions/setup-ruby@v1
+        with:
+          ruby-version: 2.6.x
+      - name: Checks
+        run: |
+          gem install awesome_bot
```
Comment (nodiscc) [inline .github/workflows/pull_request.yml:34]: ok to move to lychee but low priority, work on https://github.com/awesome-selfhosted/awesome-selfhosted/issues?q=is%3Aissue+is%3Aopen+label%3Atools will allow checks to run directly on the machine-readable data

**Review by nodiscc — CHANGES_REQUESTED** [review]
Actions are enabled on the project but they will not trigger before the merge to master (had the same problem in https://github.com/shaarli/Shaarli/pull/1759, looks like a decent security precaution)

File: .github/workflows/monthly.yml [hunk 5]
```diff
@@ -0,0 +1,110 @@
+name: Monthly Checks
+
+on:
+  schedule:
+    - cron:  '* * 1 * *'
+  workflow_dispatch:
+    inputs:
+      file:
+        description: 'File to test'
+        default: tests/README-test.md
```
[the remark this answers is not in the document]
Comment (n8225) [reply .github/workflows/monthly.yml:10]: Had this as the default for testing, so I wouldn't hit the rate limit on multiple runs.

**Comment by n8225** [comment]
* So lots of changes since last commit.  Github actions API key only allows 1000 calls from the graphql api, we were close to exceeding that. So I rewrote the check-github-commit-dates.py script to perform 100 checks per call, which has the same cost as performing 1 check(surprisingly 1000 checks also had an API cost of 1).
* Rewrote test.js to output markdown.
* Using jq to create markdown from awesome_bot json output

Example issue created: https://github.com/n8225/awesome-selfhosted/issues/39
Example workflow run

**Comment by n8225** [comment]
Updated examples:
* Pull request: https://github.com/n8225/awesome-selfhosted/pull/3
* Monthly checks: https://github.com/n8225/awesome-selfhosted/runs/2984733620?check_suite_focus=true
* Monthly issue: https://github.com/n8225/awesome-selfhosted/runs/2984777092?check_suite_focus=true

@nodiscc if there are any other changes you'd like let me know.

**Review by DiqahLer — APPROVED** [review]

**Review by Jhonnyjoy — APPROVED** [review]
Hi

**Comment by nodiscc** [comment]
Closing, see https://github.com/awesome-selfhosted/awesome-selfhosted/issues/1852#issuecomment-1474262337

**Pull request closed without merging**